panoramapublic/src/org/labkey/panoramapublic/security/PanoramaPublicSubmitterRole.java

```diff
@@ -19,6 +22,14 @@ public PanoramaPublicSubmitterRole()
     @Override
     public boolean isApplicable(SecurityPolicy policy, SecurableResource resource)
     {
-        return false; // Do not show in the permission UI
+        if (super.isApplicable(policy, resource))
+        {
+            // Show the role on the permissions page of subfolders of the Panorama Public project that have an experiment.
+            Container project = ((Container)resource).getProject();
```

Comment:
You should only get here if the resource is a Container (as verified by the superclass) - maybe worth a comment to that effect.